src/datasets/table.py

```diff
@@ -521,30 +528,48 @@ def _concat_blocks_horizontally_and_vertically(blocks: List[List[TableBlock]]) -
             tables_to_concat_vertically.append(combined_table)
         return pa.concat_tables(tables_to_concat_vertically)
 
+    @classmethod
+    def _consolidate_blocks(cls, blocks, axis: Optional[int] = None):
+        if axis is not None:
+            consolidated_blocks = []
+            for is_in_memory, block_group in groupby(blocks, key=lambda x: isinstance(x, InMemoryTable)):
```

Comment:
`groupby` won't take the order of the blocks into account.
While we don't care of the order for axis=1, we actually do want to keep the right order for axis 0.

That means that when adding a new blocks in axis 0, we just need to check whether both the new block and the last block are InMemoryTable objects and merge them.

Reply:
But `groupby` does keep the order of the blocks...

Reply:
```ipython
In [4]: for k, v in groupby([1, 2, 2, 1, 1, 1]):
   ...:     print(f"{k}: {list(v)}")
   ...:
1: [1]
2: [2, 2]
1: [1, 1, 1]

```

Reply:
Oh right ! That's perfect then :)